sdk/storage/azure-storage-blob-nio/src/test/java/com/azure/storage/blob/nio/APISpec.groovy

```diff
@@ -481,10 +488,45 @@ class APISpec extends Specification {
         }
     }
 
-    def rootToContainer(String root) {
+    def rootNameToContainerName(String root) {
         return root.substring(0, root.length() - 1)
     }
 
+    def rootNameToContainerClient(String root) {
+        return primaryBlobServiceClient.getBlobContainerClient(rootNameToContainerName(root))
+    }
+
+    def getNonDefaultRootDir(FileSystem fs) {
+        for (Path dir : fs.getRootDirectories()) {
+            if (!dir.equals(((AzureFileSystem) fs).getDefaultDirectory())) {
+                return dir.toString()
+            }
+        }
+        throw new Exception("File system only contains the default directory");
+    }
+
+    def getDefaultDir(FileSystem fs) {
+        return ((AzureFileSystem) fs).getDefaultDirectory().toString()
+    }
+
+    def getPathWithDepth(int depth) {
+        def pathStr = ""
+        for (int i = 0; i < depth; i++) {
+            pathStr += generateBlobName() + AzureFileSystem.PATH_SEPARATOR
+        }
+        return pathStr
+    }
+
+    def putDirectoryBlob(BlockBlobClient blobClient) {
+        blobClient.commitBlockListWithResponse(Collections.emptyList(), null,
+            [(AzureFileSystemProvider.DIR_METADATA_MARKER): "true"], null, null, null, null)
+    }
+
+    def checkBlobIsDir(BlobClient blobClient) {
+        return blobClient.getPropertiesWithResponse(null, null, null).getValue().getMetadata()
```

Comment:
You could also make sure the value here is true just to prevent accidental fake news. One day the service could decide they want to set this to false for non-dir like blobs